archive_updater_test.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+import subprocess
+import os
+import tempfile
+import shutil
+from archive_updater import update, fill_message_directory
+from unittest import mock
+
+class ArchiveUpdaterUpdateTest(unittest.TestCase):
+
+    @mock.patch.object(subprocess, 'Popen')
+    def test_update_success(self, mock_popen):
+        mock_popen.return_value.communicate.return_value = (b'updated', b'')
+        self.assertTrue(update(''))
+    
+    @mock.patch.object(subprocess, 'Popen')
+    def test_update_failure(self, mock_popen):
+        mock_popen.return_value.communicate.return_value = (b'failed to fetch', b'no internet connection')
+        self.assertFalse(update(''))
+
+class ArchiveUpdaterFillMessageDirectoryTest(unittest.TestCase):
+        
+    def setUp(self):
+        self.tmp_dir = tempfile.mkdtemp()
+    def tearDown(self):
+        shutil.rmtree(self.tmp_dir)
+        
+    @mock.patch.object(subprocess, 'Popen')
+    @mock.patch.object(subprocess, 'call')
+    def test_fill_message_directory_success(self, mock_call, mock_popen):
+        def writeFile(args, stdout):
+            if 'show' in args:
+                stdout.write('SUCCESS')
+                return
+            stdout.write('FAIL')
+        mock_call.side_effect = writeFile
+        mock_popen.return_value.communicate.return_value = [b'success1 success2 success3', '']
+        
+        last_used_hash = fill_message_directory('', self.tmp_dir, '')
+        
+        self.assertEqual(len(os.listdir(self.tmp_dir)), 3)
+        for filename in os.listdir(self.tmp_dir):
+            with open(os.path.join(self.tmp_dir,filename), "r") as file:
+                val = file.read()
+                self.assertEqual(val, 'SUCCESS')
+        self.assertEqual(last_used_hash,'success1')
```

Comment:
minor nit: missing spaces after `,` in the `assertEquals()` calls for `last_used_hash` across test cases